x-pack/plugins/upgrade_assistant/server/routes/ml_snapshots.ts

```diff
@@ -330,8 +331,25 @@ export function registerMlSnapshotRoutes({
       path: `${API_BASE_PATH}/ml_upgrade_mode`,
       validate: false,
     },
-    versionCheckHandlerWrapper(async ({ core }, request, response) => {
+    versionCheckHandlerWrapper(async ({ core, ...rest }, request, response) => {
       try {
+        /**
+         * Always return false if featureSet.mlSnapshots is set to false
+         * This disables showing the ML deprecations in the UA UI
+         * 
+         * This config should be set to false only on the `x.last` versions
+         * ML `upgrade_mode` can be enabeld from outside Kibana, the prurpose
+         * of this feature guard is to hide all ML related deprecations from the end user 
+         * until the next major upgrade.
+         */
```

Comment:
```suggestion
        /**
         * Always return false if featureSet.mlSnapshots is set to false
         * This disables possibly showing a needless warning about ML
         * upgrade mode when there's never a need to upgrade ML job
         * snapshots in minor version upgrades.
         * 
         * This config should be set to false only on the `x.last` versions, or when
         * the constant `MachineLearningField.MIN_CHECKED_SUPPORTED_SNAPSHOT_VERSION`
         * is incremented to something higher than 7.0.0 in the Elasticsearch code.
         */
```

I made some suggestions to edit the comment.

Setting `mlUpgradeModeEnabled` to `false` will stop https://github.com/elastic/kibana/blob/47ebe8bef760828567aaa167c7e20fc846c10a9d/x-pack/plugins/upgrade_assistant/public/application/components/es_deprecations/deprecation_types/ml_snapshots/flyout.tsx#L215 from being shown, which is good since that message is irrelevant until we actually want to start doing ML snapshot upgrades again.

However, `mlUpgradeModeEnabled` set to `false` _will_ allow actual ML snapshot deprecation messages to be shown - see https://github.com/elastic/kibana/blob/47ebe8bef760828567aaa167c7e20fc846c10a9d/x-pack/plugins/upgrade_assistant/public/application/components/es_deprecations/deprecation_types/ml_snapshots/flyout.tsx#L246. So what stops that happening? The answer is https://github.com/elastic/elasticsearch/blob/34ae3dd5b13e01878e07ecd494d0be2c92676a8f/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/MachineLearningField.java#L50.

So I think everything should work with this code change as it is, but it's just that the explanation for _why_ it will work needs clarifying.

When we want to enable ML model snapshot deprecation warnings again we need to change https://github.com/elastic/elasticsearch/blob/34ae3dd5b13e01878e07ecd494d0be2c92676a8f/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/MachineLearningField.java#L50 and also `featureSet.mlSnapshots` in the same release.

Reply:
> Do you think it makes sense to add this logic on top as well?

Yes, it's probably safest to do this.